web/api/v1/api.go

```diff
@@ -170,16 +178,18 @@ func (api *API) Register(r *route.Router) {
 	wrap := func(f apiFunc) http.HandlerFunc {
 		hf := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			setCORS(w)
-			data, err, finalizer := f(r)
-			if err != nil {
-				api.respondError(w, err, data)
-			} else if data != nil {
-				api.respond(w, data)
+			result := f(r)
+			if result.err != nil {
+				api.respondError(w, result.err, result.data)
+			} else if result.warnings != nil {
+				api.respondWarnings(w, result.warnings, result.data)
```

Comment:
Can this be handled by respond rather than having to handle warnings separately?